### ISSUE TITLE
ipython console bug at startup

### ISSUE THREAD

**Issue by benjello** [spyder-ide/spyder#2770]
## Description

IPython console unable to connect to kernel (never getting much far than the starting "connecting to Kernel)
at startup after a upgrading my debian testing box

**What steps will reproduce the problem?**

1. start spyder

**What is the expected output? What do you see instead?**

Clean run

**Please provide any additional information below**

## Version and main components

* Spyder Version: 2.3.5.2 
* Python Version: 2.7.10+
* Qt Versions:  4.8.7, PyQt4 (API v2) 4.11.4 on Linux

## Optional dependencies
```
IPython >=1.0   :  2.3.0 (OK)
jedi >=0.8.1    :  0.9.0 (OK)
matplotlib >=1.0:  1.5.0rc2 (OK)
pandas >=0.13.1 :  0.17.0+git8-gcac4ad2 (OK)
pep8 >=0.6      :  1.6.2 (OK)
psutil >=0.3    :  2.2.1 (OK)
pyflakes >=0.5.0:  1.0.0 (OK)
pygments >=1.6  :  2.0.1 (OK)
pylint >=0.25   :  1.4.4 (OK)
rope >=0.9.2    :  0.10.2 (OK)
sphinx >=0.6.6  :  1.3.1 (OK)
sympy >=0.7.3   :  0.7.6 (OK)
zmq >=2.1.11    :  14.4.0 (OK)

```

**Comment by alexendy** [comment]
Same issue here.

I initially got the problem with the packaged Debian Sid software (spyder 2.3.6 and ipython 2.3.0), but I uninstalled+purged the debian ipython, qtconsole and spyder packages and reinstalled everything through pip (so I now have spyder 2.3.7 and a ipython 4.0) and I still gets the same error.

ipython works flawlessly when used in a standalone way. The underlying Python is still the Debian Sid packaged version (2.7.10+)

**Comment by benjello** [comment]
@alexendy : did you report the bug to debian ?

**Comment by alexendy** [comment]
@benjello : nope, since after all it did not look like something specific to the debian packaged version. Maybe I should too, though.

**Comment by benjello** [comment]
@alexendy : i contacted the debian maintainer but i did not report formally the bug since i am not used to reportbug (and i do not know how to distinguish betwenn the debian python packages and those installed only via pip).

**Comment by benjello** [comment]
BTW, @alexendy did you find a way to get back to a working spyder in debian testing ?

**Comment by alexendy** [comment]
On my system, Debian-installed packages are in /usr/lib/python2.7 and pip-installed ones in /usr/local/lib/python2.7 I usually try to use debian packages as much as possible except when they don't exist or are really outdated.

I tried to investigate if it could be a duplicate or related to #2485 (the packaged Debian version of zmq seems to be possibly problematic), uninstalled de debian packaged zmq and built the latest (fixed) version, but it did not change anything.

@benjello : no, alas. I have spent some time looking for a workaround but found none yet. The best I can manage is using spyder for python3 (which, amazingly, is not concerned by the problem) where I can, and stick with standalone ipython the rest of the time (not all of my code is python3-compatible).

**Comment by ccordoba12** [comment]
I'm sorry but I can't reproduce this error :-s

And Linux is my default OS ;-)

**Comment by alexendy** [comment]
@ccordoba12 which GNU/Linux ? I suspect this is highly dependant on the distribution and version.

I have not tried to reproduce the bug for a while, but I have seen there were an update to the ipython package in Debian Sid a few weeks ago. Will try to give it a try and update soon.

**Comment by benjello** [comment]
Using matplotlib 1.4.2 instead of 1.5 solves the problem. Since it is related to #2771, I close this issue.

**Comment by ccordoba12** [comment]
I'll fix this problem in Spyder 2.3.8.

**Comment by alexendy** [comment]
Thanks a lot for the fix !
